src/github.com/getlantern/errors/errors.go

```diff
@@ -0,0 +1,641 @@
+/*
+Package errors defines error types used across Lantern project.
+
+  errors.ReportTo(myErrorReporter)
+  ...
+  if n, err := Foo(); err != nil {
+    errors.Wrap(err).Report()
+  }
+
+Wrap() method will try as much as possible to extract details from the error
+passed in, if it's errors defined in Go standard library. For application
+defined error type, at least the Go type name and what err.Error() returns
+will be recorded.
+
+Extra fields can be chained in any order, at any time.
+
+  func Foo() *Error {
+  	//...
+    return errors.New("some error").With("some_counter", 1).WithOp("connect")
```

Comment:
What you refer to as "reporter" here is, if I understand correctly, needs to be connected to what I've denominated the Event Emitter, in the events-pipeline library. This one will receive a Key and a list of values in the form `map[string]interface{}`. But the actual interface between your error reporter and the my emitter would convert the static types to a `map[string]interface{}`. Does that sound reasonable to you?